src/main/java/com/ibm/crypto/plus/provider/XDHPublicKeyImpl.java

```diff
@@ -411,7 +411,9 @@ public byte[] getEncoded() {
                 oidSeq.putOID(this.algid.getOID());
                 if ((oidSubSeq != null)) {
                     oidSeq.write(DerValue.tag_Sequence, oidSubSeq);
-                } else {
+                }
```

Comment:
According to the previous code, the logic is, if `oidSubSeq == null`, then `oidSeq.putNull();`. And this `oidSubSeq` will only be `null` if `CurveUtil` is not `FFDHE`. If the `CurveUtil` is `FFDHE`, then `oidSubSeq != null`, then it won’t `oidSeq.putNull();`.

But according to the test and also the comment in the codes, this issue is the `XDH` format changes between JDK11 or before and JDK17 or after, no matter the `Curve` is `FFDHE` or not.

So, right now, the current logic is, no matter is `FFDHE` or not, if the JDK version is JDK11 or before, then `oidSeq.putNull();`, if the JDK version is JDK17 or after, then don’t `oidSeq.putNull();`.

Reply:
Yeah, but the difference should be there if the curve is not `FFDHE`, in which case you add `NULL` in `Java <=11` and nothing otherwise. I don't think you want to add `NULL` if the curve is `FFDHE`, which is what will happen now if `Java <=11`, where you will have both the `oidSubSeq` and `NULL`.

Reply:
As I understand it, we would need something like:
```
if ((oidSubSeq != null)) {
   oidSeq.write(DerValue.tag_Sequence, oidSubSeq);
} else if (Integer.parseInt(provider.getVersionStr()) <= 11) {
   oidSeq.putNull();
}
```
Unless, of course, there is something else I'm missing.

Reply:
Right, agree, updated the code.